MovingBlocks/Terasology
# Continuing work on nodes:  DeferredPointLightsNode and a few other things

### this PR:
- refactors the LightGeometryNode into DeferredPointLightsNode, to move it to the new architecture
- refactors DirectionalLightNode for further work by isolating it from the DeferredPoingLightsNode.
- added some method/class-level javadocs
- fixes problem in OverlaysNode  - see first commit description
- updates WorldRendererImpl to account for the work done, i.e. the newly named node.

### How to test
- Run Terasology, 
- start a saved game
- make sure there is nothing unusual with the image shown on screen.
